commonv5/ts/sonarqube/__tests__/TaskReport-test.ts

```diff
@@ -135,6 +135,62 @@ it("should find report files for SonarQube above 7.2.0", async () => {
   expect(tl.findMatch).toHaveBeenCalledWith("mock root search path", expectedSearchPath);
 });
 
+it("should find report files for SonarQube above 10.0", async () => {
+  // using spyOn so we can reset the original behaviour
+  jest.spyOn(tl, "getVariable").mockImplementation(() => "mock root search path");
+  jest.spyOn(tl, "findMatch").mockImplementation(() => ["path1", "path2"]);
+
+  const endpoint = new Endpoint(EndpointType.SonarQube, null);
+
+  const reportFiles = await TaskReport.findTaskFileReport(endpoint, new semver.SemVer("10.0.0"));
+
+  expect(reportFiles.length).toBe(2);
+  expect(reportFiles[0]).toBe("path1");
+  expect(reportFiles[1]).toBe("path2");
+
+  expect(tl.getVariable).toHaveBeenCalledTimes(2);
+  expect(tl.getVariable).toBeCalledWith("Agent.TempDirectory");
+
+  // Calculate the expected path to take account of different
+  // path separators in Windows/non-Windows
+  const expectedSearchPath = path.join(
+    "sonar",
+    tl.getVariable("Build.BuildNumber"),
+    "**",
+    "report-task.txt"
+  );
+  expect(tl.findMatch).toHaveBeenCalledTimes(1);
+  expect(tl.findMatch).toHaveBeenCalledWith("mock root search path", expectedSearchPath);
+});
+
+it("should find report files for SonarQube above 20.0", async () => {
+  // using spyOn so we can reset the original behaviour
+  jest.spyOn(tl, "getVariable").mockImplementation(() => "mock root search path");
+  jest.spyOn(tl, "findMatch").mockImplementation(() => ["path1", "path2"]);
+
+  const endpoint = new Endpoint(EndpointType.SonarQube, null);
+
+  const reportFiles = await TaskReport.findTaskFileReport(endpoint, new semver.SemVer("20.0.0"));
+
+  expect(reportFiles.length).toBe(2);
+  expect(reportFiles[0]).toBe("path1");
+  expect(reportFiles[1]).toBe("path2");
+
+  expect(tl.getVariable).toHaveBeenCalledTimes(2);
+  expect(tl.getVariable).toBeCalledWith("Agent.TempDirectory");
+
+  // Calculate the expected path to take account of different
+  // path separators in Windows/non-Windows
+  const expectedSearchPath = path.join(
+    "sonar",
+    tl.getVariable("Build.BuildNumber"),
+    "**",
+    "report-task.txt"
+  );
+  expect(tl.findMatch).toHaveBeenCalledTimes(1);
+  expect(tl.findMatch).toHaveBeenCalledWith("mock root search path", expectedSearchPath);
+});
```

Comment:
Why this test? What value does it bring compared to the identical one above? ("should find report files for SonarQube above 10.0")

Reply:
Yes the idea was to test few other major version to make sure we will be ok. Thanks for the tip, I applied it.